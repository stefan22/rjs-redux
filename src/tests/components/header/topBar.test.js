import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import TopBar from '../../../components/header/top-bar';


describe('TopBar component', () => {
  it('should render top navigation correctly', () => {
    const wrapper = shallow(<TopBar />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should have four page-links, each with its own class', () => {
    const wrapper = shallow(<TopBar />);
    expect(wrapper.find('.nav.topbar-home').length).toBe(1);
    expect(wrapper.find('.nav.topbar-expenses').length).toBe(1);
    expect(wrapper.find('.nav.topbar-movies').length).toBe(1);
    expect(wrapper.find('.nav.topbar-earthquakes').length).toBe(1);
  });

});

