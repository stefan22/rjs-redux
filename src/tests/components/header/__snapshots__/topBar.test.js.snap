// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component should render top navigation correctly 1`] = `
<div
  className="topbar"
>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        className="topbar-icon"
        color="inherit"
      >
        <img
          alt="docs-logo"
          src="docs.png"
          width="500"
        />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        className="topbar-grow"
        color="inherit"
        variant="h6"
      />
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        className="nav topbar-home"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        className="nav topbar-expenses"
        to="/expenses"
      >
        Expenses
      </NavLink>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        className="nav topbar-movies"
        to="/movies"
      >
        Movies
      </NavLink>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        className="nav topbar-earthquakes"
        to="/earthquakes"
      >
        Earthquakes
      </NavLink>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
